File: javascript/e3-runner-node/src/runner.ts
#!/usr/bin/env node

/**
 * E3 Node.js Runner
 *
 * Watches the queue/node/ directory for tasks and executes them
 * using the Node.js runtime with the east-node platform.
 */

import * as fs from 'fs';
import * as path from 'path';

const E3_REPO = process.env.E3_REPO || path.join(process.env.HOME || '', '.e3');
const QUEUE_DIR = path.join(E3_REPO, 'queue', 'node');

console.log(`E3 Node.js Runner starting...`);
console.log(`Repository: ${E3_REPO}`);
console.log(`Queue: ${QUEUE_DIR}`);

// TODO: Implement runner
// - Watch QUEUE_DIR for new task files (using fs.watch)
// - Atomically claim tasks (rename with worker ID)
// - Load task commit and check for memoization
// - Execute tasks with logging
// - Store results and create completion commits
// - Handle child task spawning

process.on('SIGINT', () => {
  console.log('\nShutting down...');
  process.exit(0);
});

console.log('Watching for tasks...');

File: python/e3-runner-python/e3_runner/runner.py
#!/usr/bin/env python3
"""E3 Python Runner.

Main entry point for the Python task runner.
"""

import os
import sys
from pathlib import Path


def main() -> None:
    """Run the Python task runner."""
    e3_repo = os.environ.get("E3_REPO", str(Path.home() / ".e3"))
    queue_dir = Path(e3_repo) / "queue" / "python"

    print("E3 Python Runner starting...")
    print(f"Repository: {e3_repo}")
    print(f"Queue: {queue_dir}")

    # TODO: Implement runner
    # - Watch queue_dir for new task files (using watchdog)
    # - Atomically claim tasks (rename with worker ID)
    # - Load task commit and check for memoization
    # - Execute tasks with logging
    # - Store results and create completion commits
    # - Handle child task spawning

    print("Watching for tasks...")

    try:
        # Keep running
        import time

        while True:
            time.sleep(1)
    except KeyboardInterrupt:
        print("\nShutting down...")
        sys.exit(0)


if __name__ == "__main__":
    main()
